jetty-http2/http2-http-client-transport/src/test/java/org/eclipse/jetty/http2/client/http/MultiplexedConnectionPoolTest.java

```diff
@@ -0,0 +1,306 @@
+        startServer(new EmptyServerHandler()
+        {
+            @Override
+            protected void service(String target, Request jettyRequest, HttpServletRequest request, HttpServletResponse response) throws ServletException
+            {
+                if (!target.equals("/block"))
+                    return;
+
+                handlerSignalingSemaphore.release();
+
+                try
+                {
+                    handlerWaitingSemaphore.acquire();
+                }
+                catch (Exception e)
+                {
+                    throw new ServletException(e);
+                }
+            }
+        });
+
+        QueuedThreadPool clientThreads = new QueuedThreadPool();
+        clientThreads.setName("client");
+        HttpClientTransport transport = new HttpClientTransportOverHTTP2(new HTTP2Client());
+        transport.setConnectionPoolFactory(factory.factory);
+        client = new HttpClient(transport, null);
+        client.setExecutor(clientThreads);
+        client.setIdleTimeout(maxIdle);
+        client.start();
+
+        CountDownLatch latch1 = new CountDownLatch(1);
+        CountDownLatch latch2 = new CountDownLatch(2);
+        // create 2 requests that are going to consume all the multiplexing slots
+        clientThreads.execute(() -> client.newRequest("http://localhost:" + connector.getLocalPort() + "/block")
+            .send(result ->
+            {
+                if (result.isSucceeded())
+                {
+                    latch1.countDown();
+                    latch2.countDown();
+                }
+            }));
+
+        // wait for the 1st request to be serviced to make sure only 1 connection gets created
+        handlerSignalingSemaphore.acquire();
+
+        clientThreads.execute(() -> client.newRequest("http://localhost:" + connector.getLocalPort() + "/block")
+            .send(result ->
+            {
+                if (result.isSucceeded())
+                {
+                    latch1.countDown();
+                    latch2.countDown();
+                }
+            }));
+
+        // wait for both requests to start being serviced
+        handlerSignalingSemaphore.acquire();
+
+        assertThat(poolCreateCounter.get(), is(1));
+
+        // finalize 1 request, freeing up 1 multiplexing slot
+        handlerWaitingSemaphore.release();
+        // wait until 1st request finished
+        assertTrue(latch1.await(5, TimeUnit.SECONDS));
+
+        assertThat(poolRef.get().getInUseCount(), is(1));
+        assertThat(poolRef.get().getIdleCount(), is(0));
+        assertThat(poolRef.get().getClosedCount(), is(0));
+        assertThat(poolRef.get().size(), is(1));
+
+        // wait for the connection to expire
+        Thread.sleep(maxDuration + 500);
+
+        // send a 3rd request that will close the expired multiplexed connection
+        ContentResponse response = client.newRequest("localhost", connector.getLocalPort())
```

Comment:
Be explicit and add `.path("/dontblock")`.

Reply:
Ok.